client/app/components/navbar/navbar.service.js

```diff
@@ -0,0 +1,24 @@
+import template from './navbar-menu-header/navbar-menu-header.component.html';
+
+{
+  class Navbar {
+    constructor($compile, $rootScope, $timeout) {
+      this.$compile = $compile;
+      this.$rootScope = $rootScope;
+      this.$timeout = $timeout;
+    }
+
+    buildMenuHeader(content) {
+      const compiledTemplate = this.$compile(template)(angular.merge(
```

Comment:
```suggestion
      const compiledTemplate = this.$compile(template)({
      ...this.$rootScope.$new(true),
      { $ctrl: { content } }
      })
```
?

Reply:
This won't work as it seems `prototype` is not copied